src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/FindAllReferencesHandler.cs

```diff
@@ -122,8 +122,13 @@ public async Task<VSReferenceItem[]> HandleRequestAsync(ReferenceParams request,
                     cancellationToken).ConfigureAwait(false);
 
                 if (mappingResult == null ||
-                    mappingResult.HostDocumentVersion != documentSnapshot.Version ||
-                    mappingResult.Ranges[0].IsUndefined())
+                    mappingResult.Ranges[0].IsUndefined() ||
+
+                    // Only verify HostDocumentVersion when the FAR request was initiated from the same
+                    // document as the reference item. This assumes that background document changed while this
+                    // request was performed. We could fetch other razor docs with the DocumentManager however
```

Comment:
> We could fetch other razor docs with the DocumentManager

We can only do that for open documents. Closed documents are not tracked by the LSPDocumentManager. Also, I don't think there is any performance penalty in fetching documents from the DocumentManger now that we aren't UI thread dependent. See [my fix](https://github.com/dotnet/aspnetcore-tooling/pull/1993) for the same issue with go to def and friends.

Reply:
>Closed documents are not tracked by the LSPDocumentManager

Ah I see, so we can't use it to find the versioning state of closed documents (which makes sense as if it's closed, we could assume it's current).

>See my fix for the same issue with go to def and friends.
```csharp
if (mappingResult == null ||
                (_documentManager.TryGetDocument(razorDocumentUri, out var documentSnapshot) &&
                    mappingResult.HostDocumentVersion != documentSnapshot.Version))
```
So as it's possible for the mapped reference results to include closed files, we wouldn't be able to perform this check. Are you suggesting trying the version check for all the open files, and skipping it for closed files?

Reply:
You can, only do the check if the file is open and if it is closed assume versions are in-sync 😄.

Reply:
Makes sense, thanks for the clarification. Updated 😄